### PR TITLE
Network calls

### PR DESCRIPTION
**Pull request by godfreaherald**
Added network calls using Fetch and async/await construct to get data from a remote server and display on . the booksList page. When data is loading and animation to let the user know is set.

### CODE REVIEW

File: lesson-2/src/components/TopNav/TopNav.js [hunk 1]
```diff
@@ -1,19 +1,18 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
-import { getUser } from '../../services/index';
-
+//import { getUser } from '../../services/index';
```
Comment (alfredayibonte) [inline lesson-2/src/components/TopNav/TopNav.js]: kindly, remove any code that is not needed

File: lesson-2/src/pages/books/BookList.js [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+  const [error, setErrors] = useState('');
+  useEffect(() => {
+    setLoading(true);
+    fetchData();
+  }, []);
+
+  const fetchData = async () => {
+    try {
+      setBooks(await bookList());
+      //  bookList().then(books => setBooks(books));
```
Comment (alfredayibonte) [inline lesson-2/src/pages/books/BookList.js]: Kindly, remove any code not needed

File: lesson-2/src/pages/books/components/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+const BooksData = ({ booklist }) => {
+  console.log(booklist);
```
Comment (alfredayibonte) [inline lesson-2/src/pages/books/components/BooksList.js]: Kindly, remove all logs as well

File: lesson-2/src/pages/layout/Contants.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export const HOME = '/';
+export const LOGIN = '/login';
+export const NOT_FOUND = '/403';
+export const SHOP = '/shop';
+export const BOOKS = '/books';
+export const LOGOUT = '/logout';
+export const BOOKLIST = 'http://www.mocky.io/v2/5e3d13222d00005100d95a98';
```
Comment (alfredayibonte) [inline lesson-2/src/pages/layout/Contants.js]: This name is a bit misleading. Is it possible to change to perhaps BOOKS_ENPOINT or BOOKS_URL?

File: lesson-2/src/pages/layout/index.js [hunk 5]
```diff
@@ -11,45 +11,36 @@ import Shop from '../shop/shop';
 import BookList from '../books/BookList';
 import BookShow from '../books/BookShow';
 import SignInPage from '../signin/signin';
-import { getUser, logoutUser } from '../../services/index';
+import { getUser, logoutUser } from '../../services';
+import PageNotFoundCompoment from '../static-pages/page-not-found';
 import './index.style.scss';
+import ProtectedRoute from './shared/ProtectedRoute';
```
Comment (alfredayibonte) [inline lesson-2/src/pages/layout/index.js]: For consistency sake, you should change the file name `ProtectedRoute`  to  `protected-route`

File: lesson-2/src/pages/signin/signin.js [hunk 6]
```diff
@@ -1,29 +1,38 @@
 import React, { useState } from 'react';
 import { Redirect } from 'react-router-dom';
 import { authenticationUser, getUser } from '../../services/index';
-const SignIn = ({ history }) => {
+import { HOME } from '../layout/Contants';
+const SignIn = ({ history, location, ...rest }) => {
+  console.log('login-location:::', location);
```
Comment (alfredayibonte) [inline lesson-2/src/pages/signin/signin.js]: Remove log here

File: lesson-2/src/pages/signin/signin.js [hunk 7]
```diff
@@ -1,29 +1,38 @@
...
   const [state, setState] = useState({});
+
   const handleChange = event => {
     const { value, name, id } = event.currentTarget;
     console.log(value, name, id);
```
Comment (alfredayibonte) [inline lesson-2/src/pages/signin/signin.js]: remove log

File: lesson-2/src/pages/books/BookList.js [hunk 8]
```diff
@@ -1,6 +1,37 @@
...
-export default BookList;
\ No newline at end of file
+import React, { useState, useEffect } from 'react';
+import { bookList } from './actions/index';
+import Loadingbooks from './components/Loadingbooks';
+import BooksError from './components/BooksError';
+import BooksData from './components/BooksList';
+
+const BookList = props => {
+  //console.log('booklist', props);
```
Comment (alfredayibonte) [inline lesson-2/src/pages/books/BookList.js]: remove comment

File: lesson-2/src/pages/books/BookList.js [hunk 9]
```diff
@@ -1,6 +1,36 @@
...
+    } catch (error) {
+      setError(error);
+    }
+  };
+
+  return (
+    <div>
+      <h2>Book List</h2>
+
+      {<Loadingbooks loading={loading} />}
```
Comment (alfredayibonte) [inline lesson-2/src/pages/books/BookList.js]: why are the components in curl braces `{}`?

File: lesson-2/src/pages/home/home.js [hunk 10]
```diff
@@ -1,6 +1,6 @@
-import React from 'react'
-const Home = (props) => {
-    console.log('home:::', props)
-    return <div>I am in home</div>
-}
-export default Home;
\ No newline at end of file
+import React from 'react';
+const Home = props => {
+  console.log('home:::', props);
```
Comment (alfredayibonte) [inline lesson-2/src/pages/home/home.js]: Kindly, remove log

**Comment by godfreaherald** [comment]
done the editing.

**Pull request merged**